src/Sparrow/Json/JsonDeserializationBase.cs

```diff
@@ -375,6 +375,9 @@ private static ParameterExpression GetParameter(Type type, Dictionary<Type, Para
         private static Dictionary<string, T> ToDictionaryOfPrimitive<T>(BlittableJsonReaderObject json, string name, JsonDeserializationStringDictionaryAttribute jsonDeserializationDictionaryAttribute)
             where T : struct
         {
+            if (json == null)
```

Comment:
why we added all those `null` checks?

Reply:
Because you might actually get nulls here, and we do when working with null topology, for example.
This is deserialization code and should handle this scenario.